### PR TITLE
Feature/dx safe provider

### PR DESCRIPTION
**Pull request by giacomolicari**
The PR applies the following changes:
- Merge feature/multi-safe into feature /dx-safe-provider
- Add SafeModule integration test

### CODE REVIEW

File: conf/config-web3.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    blockForNonceCalculation: 'pending',
+    safes: [{
+      operatorAddressIndex: 0,
+      safeAddress: process.env.SAFE_ADDRESS,
+      safeModuleType: process.env.SAFE_MODULE_TYPE,
+      safeModuleAddress: process.env.SAFE_MODULE_ADDRESS
+    }]
+  }
+
+  moduleExport = {
```
Comment (anxolin) [inline conf/config-web3.js]: Why do we need to conditionally export the config?

It's not enough to export the WEB3_PROVIDER?

File: conf/index.js [hunk 2]
```diff
@@ -12,7 +12,8 @@ const defaultConf = {
   ...require('./config-bots'),
   ...require('./config-contracts'),
   ...require('./config-notification'),
-  ...require('./config-repos')
```
Comment (anxolin) [inline conf/index.js]: I don't understand why is highlighting this change. It was done in the branch you are trying to merge.
There's more cases like this, not sure why it happens.

File: package.json [hunk 3]
```diff
@@ -26,11 +26,12 @@
     "bots-rinkeby-dev": "DEBUG=ERROR-*,WARN-*,INFO-*,DEBUG-* PUBLIC_API_PORT=9091 NETWORK=rinkeby node src/runBots.js",
     "bots-kovan": "PUBLIC_API_PORT=9091 NETWORK=kovan node src/runBots.js",
     "bots-mainnet": "MARKETS=RDN-WETH RDN_TOKEN_ADDRESS=0x255aa6df07540cb5d3d297f0d0d4d84cb52bc8e6 OMG_TOKEN_ADDRESS=0xd26114cd6EE289AccF82350c8d8487fedB8A0C07 PUBLIC_API_PORT=9092 NETWORK=mainnet node src/runBots.js",
-    "test": "NODE_ENV=test jest",
-    "test-coverage": "NODE_ENV=test jest --coverage && cat ./coverage/lcov.info | coveralls",
+    "test": "NODE_ENV=test jest --testPathIgnorePatterns tests/safe-module",
```
Comment (anxolin) [inline package.json]: The ignores I would configure them using jest.config.js so it works also if you execute `npx jest ...`

File: conf/config-web3.js (hunk 1 above)
Comment (giacomolicari) [reply conf/config-web3.js]: It's required here https://github.com/gnosis/dx-services/blob/feature/multi-safes/src/helpers/web3Providers/index.js#L15

File: conf/index.js (hunk 2 above)
Comment (giacomolicari) [reply conf/index.js]: It's highlighted because we added a ',' and a further require() on line 16

File: package.json (hunk 3 above)
Comment (giacomolicari) [reply package.json]: Yes, that would be great :)

**Comment by anxolin** [comment]
Good job Giacomo!

**Pull request merged**